universe/Conditions.cpp

```diff
@@ -1501,14 +1501,14 @@ unsigned int Target::GetCheckSum() const {
 // Homeworld                                             //
 ///////////////////////////////////////////////////////////
 Homeworld::Homeworld() :
-    Condition(),
-    m_names()
+    Homeworld(std::vector<std::unique_ptr<ValueRef::ValueRef<std::string>>>{})
 {}
 
 Homeworld::Homeworld(std::vector<std::unique_ptr<ValueRef::ValueRef<std::string>>>&& names) :
     Condition(),
     m_names(std::move(names))
 {
+    // all_of should return true if m_names is empty
```

Comment:
Pointless comment, that behaviour is already documented in the boost range API/the described behavior according to the C++ specification.

https://www.boost.org/doc/libs/1_52_0/libs/algorithm/doc/html/algorithm/CXX11.html

> all_of and all_of_equal both return true for empty ranges, no matter what is passed to test against. When there are no items in the sequence to test, they all satisfy the condition to be tested against.

Reply:
It's not obvious from the name `all_of` and I initially guessed the problem was that it should be `m_names.empty() || boost::algorithm::all_of(...)`